### ISSUE TITLE
Regarding reports received recently and would like some clearification

### ISSUE THREAD

**Issue by JixunMoe** [JasonBarnabe/greasyfork#671]
## Combined scripts (multi-purpose)

Some of those looks like a combination of scripts running on different sites, with different purposes.

I think the purpose is to have a long name / description that can show up from different search terms, so people will install their scripts and they get paid (via automated affiliate links).

Personally I don't like those scripts, as they are hard to audit (too long) and making all kinds of cross domain requests. I need some advice on handling reports regarding those.

(Also, they tend to put a lot of images/text to describe their "affiliates plan", then mentions other features).

Examples:

* https://greasyfork.org/scripts/391865
* https://greasyfork.org/scripts/396155
* https://greasyfork.org/scripts/392216
* https://greasyfork.org/scripts/390952
* https://greasyfork.org/scripts/384538
* https://greasyfork.org/scripts/370634

## Behind  a"free"wall

Some scripts simply adds a button that redirects the user to a different site, and then the user is required to sign in via WeChat & subscribe their accounts. While the intention can be good, this can be easily abused. 

Some background about "WeChat"
> TL; DR: A social app in China that people can get paid by having people subscribe to them.
> 
> "WeChat" is an app similar to Facebook & Instagram, where people can send messages and post updates under their "real" identity (people tend to know each other offline). By having people subscribe to their "official account", the owner can make a profit by posting sponsored contents and/or affiliate links.

Personally, I don't like this kind of scripts behind a "free"wall. I've deleted the one that didn't declare this behaviour.

Examples:

* https://greasyfork.org/scripts/390952
* https://greasyfork.org/scripts/392216

## "Free" Proxy

People posting scripts titled SS (ShadowSocks), SSR, V2Ray or some other tooling, which gives free access to VPN. They ask its user to navigate to a specific page, then the script will inject the connection information at the page.

I believe those scripts should be rejected and instead hosted elsewhere since the site is not designed for information publishing.

Examples:

* https://greasyfork.org/scripts/400691
* https://greasyfork.org/scripts/399057
* https://greasyfork.org/scripts/400767

---

Would like to have more clear boundaries regarding those 3 points written in the https://greasyfork.org/help/code-rules page, what is allowed and what is not.

**Comment by JasonBarnabe** [comment]
These specific cases aren't called out, but seems like they are covered by the existing rules. These rules could be expanded to explicitly mention these cases.

**Combined scripts (multi-purpose)**

- Scripts must include a description of what they do and may not do things unreasonably outside of this description. Users must know what a script will do before installing it.
- Scripts with monetization methods (tracking, ads, miners, etc.) are not forbidden, but the author must disclose this prominently in the script description or additional info.

**Behind a"free"wall**

- Scripts must have a reason to be a script. For example, a script that just directs a user to a website doesn't need to be a script; it could be a link.

**"Free" Proxy**

- Scripts must have a reason to be a script. For example, a script that just directs a user to a website doesn't need to be a script; it could be a link.

---

The multi-purpose one is kind of interesting - maybe we should have something that says you can do a lot of things on one site, or one thing on a lot of sites, but not a lot of different things on a lot of sites?

**Comment by JixunMoe** [comment]
Regarding multi-purpose scripts, I would like to invite other mods to discuss as well.

My preference is:

> not a lot of different things on a lot of sites

**Comment by JasonBarnabe** [comment]
Adding:

- Scripts must not require users to perform an unrelated action, such as sending a payment, subscribing to a channel, or liking a page, for the script to be functional
- Scripts must not include `@include`s, `@match`es, or keywords for sites they do not provide functionality on.
